src/app/api/link/route.ts

```diff
@@ -0,0 +1,62 @@
+import { NextResponse } from "next/server";
+
+import { AppManifest } from "@/data";
+import { fetchAllApps } from "@/data/storage";
+
+
+export async function POST(req: Request) {
+  const res = await req.json();
+
+  if (!res.link) {
+    return NextResponse.json({ error: "Missing handle in the request body" });
+  }
+
+  const apps = await fetchAllApps();
+
+  const linkType = determineLinkTypeForAppManifests(res.link as string, apps as AppManifest[]);
+
+  return NextResponse.json({ linkType });
+}
+
+const SHARE_PROFILE_URL = "https://share.lens.xyz/u/";
+const SHARE_PUBLICATION_URL = "https://share.lens.xyz/p/";
+
+function determineLinkTypeForApp(
+  link: string,
+  manifest: AppManifest
+): { type: "profile" | "publication" | "unknown"; url?: string } {
+  const profileUrl = manifest.routes.profile?.url;
+  const publicationUrl = manifest.routes.publication?.url;
+
+  if (profileUrl) {
+    const profileRegex = new RegExp(`^${profileUrl.replace(":handle", "(.+)")}$`);
+
+    const profileMatch = profileRegex.exec(link);
+    if (profileMatch) {
+      const customProfileUrl = SHARE_PROFILE_URL + profileMatch[1];
+      return { type: "profile", url: customProfileUrl };
+    }
+  }
+
+  if (publicationUrl) {
+    const publicationIdRegex = new RegExp(`^${publicationUrl.replace(":id", "(.+)")}$`);
+
+    const publicationMatch = publicationIdRegex.exec(link);
+    if (publicationMatch) {
+      const customPublicationUrl = SHARE_PUBLICATION_URL + publicationMatch[1];
+      return { type: "publication", url: customPublicationUrl };
+    }
+  }
```

Comment:
I think we should take the opportunity we know the App to add the attribution param: `?by=<app-id>`. WDYT?